modules/standard/Random.chpl

```diff
@@ -535,11 +555,77 @@ module Random {
       compilerError("RandomStreamInterface.fillRandom called");
     }
 
+    /*
+     Returns a random sample from a given 1-D array, ``x``.
+
+     :arg x: a 1-D array with values that will be sampled from.
+     :arg size: An optional integral value specifying the number of elements to
+                choose, or a domain specifying the dimensions of the
+                sampled array to be filled, otherwise a single element will be
+                chosen.
+     :arg replace: an optional ``bool`` specifying whether or not to sample with
+                   replacement, i.e. elements will only be chosen up to one
+                   time when ``replace=false``.
+     :arg prob: an optional 1-D array that contains probabilities of choosing
+                each element of ``x``, otherwise elements will be chosen over
+                a uniform distribution. ``prob`` must have integral or real
+                element type, with no negative values and at least one non-zero
+                value. The size must be equal to that of ``x.domain``.
+
+     :return: An element chosen from ``x`` if ``size == 1``, or an array of
+              element chosen from ``x`` if ``size > 1`` or ``size`` is a
+              domain.
+
+     :throws IllegalArgumentError: if ``x.size == 0``,
+                                   if ``x.size != prob.size``,
+                                   if ``prob`` contains a negative value,
+                                   if ``prob`` has no non-zero values,
+                                   if ``size < 1 || size.size < 1``,
+                                   if ``replace=false`` and ``size > x.size || size.size > x.size``
+     */
+     proc choice(x: [], size:?sizeType=none, replace=true, prob:?probType=none) throws
+     {
+       compilerError("RandomStreamInterface.choice called");
+     }
+
+    /*
+     Returns a random sample from a given 1-D range, ``x``.
+
+     :arg x: a 1-D range with values that will be sampled from.
```

Comment:
One little nitpick I caught while looking at rendered docs: a range can only be 1 dimensional in Chapel (domains introduce dimensionality by combining multiple ranges). Maybe we should specify bounded range too, so I'd suggest:

`s/1-D range/bounded range/g` for all the `choice()` docs, e.g.

```suggestion
     :arg x: a bounded range with values that will be sampled from.
```

Reply:
> I caught while looking at rendered docs: a range can only be 1 dimensional in Chapel (domains introduce dimensionality by combining multiple ranges). Maybe we should specify bounded range too, so I'd suggest:

Yeah Agreed!

Reply:
Made the changes :-)